src/http/http-client.js

```diff
@@ -0,0 +1,223 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import { timedRequest, createCancelToken } from '../http-utils';
+import ErrorCodes from '../error-codes';
+import UploadOptionsBase from '../upload-options-base';
+import UploadError from '../upload-error';
+import HttpResponse from './http-response';
+import { v4 as uuid } from 'uuid';
+
+/**
+ * Invoked when there is an immediate retry error. Handles special cases and adds the
+ * error to the given result.
+ * @param {Error|string} e The error that occurred.
+ * @param {HttpResult} httpResult Result being provided for the current operation.
+ */
+function handleRetryError(e, httpResult) {
+    if (e && e.message && e.message === ErrorCodes.USER_CANCELLED) {
+        throw new UploadError('user cancelled the operation', ErrorCodes.USER_CANCELLED);
+    }
+    if (e && e.isAxiosError) {
```

Comment:
Warning: Leaky abstraction - you're using the fact that you know axios is the http handler.

Reply:
Since the error is a parameter, maybe it can be abstracted at error level?

Reply:
Good catch, I moved the logic to `http-utils`, which knows about `axios` already.